mapping_parenting_tech/analysis/prototyping/play_store/get_play_store_apps.py

```diff
@@ -7,141 +7,670 @@
+
+            # process and save what we've just downloaded:
+            review_fetch, existing_review_ids = _process_review_grab(
+                app_id, review_fetch, existing_review_ids
+            )
+            with open(target_file, "at", newline="") as csv_file:
+                csv_writer = csv.DictWriter(csv_file, fieldnames=FIELD_NAMES)
+                csv_writer.writerows(review_fetch)
+
+            # save a snapshot of where we are so we can resume later
+            log_info = _update_log_info(
+                app_id,
+                log_info,
+                continuation_token,
+                review_fetch,
+                first_pass=first_pass,
+            )
+            with open(log_file, "rb+") as log_file_handle:
+                pickle.dump(log_info, log_file_handle)
+
+            app_review_count += len(review_fetch)
+
+            if not run_quietly:
+                logging.info(
+                    f"Retrieved {app_review_count} new reviews for {app_id}; fetched {log_info[app_id]['downloaded']} in total"
+                )
+
+            first_pass = False
+
+        # if we've reached here and keep_going is true it's because we've downloaded all the results, so update logfile accordingly
+        if keep_going and len(review_fetch) > 0:
+            with open(log_file, "rb+") as log_file_handle:
+                log_info = _update_log_info(
+                    app_id, log_info, continuation_token, review_fetch, completed=True
+                )
+                pickle.dump(log_info, log_file_handle)
 
-logging.info(f"Retrieved {running_total} reviews")
+        if not run_quietly:
+            logging.info(f"{app_id}: done")
 
+    return True
+
+
+# %%
+def load_app_reviews(filename: str, folder: str = APP_IDS_PATH) -> list:
+    """ """
+
+    review_list = list()
+    with open(folder / filename) as csv_reader:
+        reviews = csv.DictReader(csv_reader)
+        for review in reviews:
+            review_list.append(review)
+
+    return review_list
+
+
+# %% [markdown]
+# ## Putting it together
+# ### Retrieving and using app ids via a set of downloaded web pages
+
+# %%
+app_ids = parse_folder_for_app_ids("play_store/kids_under_five")
+save_app_ids(app_ids, "kids_under_five", "kids_under_five_ids.csv")
+
+# %%
+# _, app_ids = load_app_ids("kids_under_five_ids.csv")
+app_details = get_playstore_app_details(app_ids)
+
+# %%
+save_app_details(app_details, "kids_under_five_details.json")
 
 # %%
-output_target = OUTPUT_PATH / OUTPUT_PLAY_REVIEWS_FILE
-with open(output_target, "w") as output_file:
-    json.dump(all_app_reviews, output_file, indent=2, default=str)
-output_file.close
+save_playstore_app_list_reviews(app_ids, "kids_under_five_reviews.csv")
 
-print(f"{running_total} reviews saved to {output_target}")
+# %% [markdown]
+# ### Retrieving and using app ids via apps related to a seed
+# Using `com.easypeasyapp.epappns` here
+
+# %%
+app_set = app_snowball("com.easypeasyapp.epappns", 5)
```

Comment:
This line returned the following error:
```
...
     17 
     18     snowball = set()
---> 19     snowball.update(similar_apps)
     20 
     21     if __current_depth < depth:

TypeError: 'NoneType' object is not iterable
```

I guess `.update()` only works with iterable inputs and for some reason `None` got passed there... Probably can fix the edge case it with a simple if statement?